csrc/runtime/executor.cpp

```diff
@@ -208,6 +208,15 @@ void KernelExecutor::compile(
     return;
   }
 
+  const bool has_any_cuda_tensor = std::any_of(
```

Comment:
Doing this check at the segment level instead of `runFusionWithInputs`, since the entire fusion may have a CUDA tensor input but individual segments may not. See the test case added in the PR for example.